sockets/v0/Socket.h

```diff
@@ -62,53 +63,64 @@ class Socket {
      * Start the process of resolving a domain name.
      * If the input is a text IP address, an event is queued immediately; otherwise, onDNS is
      * queued as soon as DNS is resolved.
-     * The socket must have been opened before resolve is called
+     * The socket must have been opened before resolve is called. If it hasn't been opened then
+     * returns SOCKET_ERROR_NULL_PTR
      * @param[in] address The domain name to resolve
      * @param[in] onDNS The handler to call when the name is resolved
      * @return SOCKET_ERROR_NONE on success, or an error code on failure
      */
     virtual socket_error_t resolve(const char* address, const DNSHandler_t &onDNS);
+
     /**
      * Open the socket.
      * Instantiates and initializes the underlying socket. Receive is started immediately after
      * the socket is opened.
+     * If there is no valid underlying network stack then SOCKET_ERROR_BAD_STACK is returned.
      * @param[in] af Address family (SOCKET_AF_INET4 or SOCKET_AF_INET6), currently only IPv4 is supported
      * @param[in] pf Protocol family (SOCKET_DGRAM or SOCKET_STREAM)
      * @return SOCKET_ERROR_NONE on success, or an error code on failure
      */
     virtual socket_error_t open(const socket_address_family_t af, const socket_proto_family_t pf);
+
     /**
      * Binds the socket's local address and IP.
      * 0.0.0.0 is accepted as a local address if only the port is meant to be bound.
      * The behaviour of bind("0.0.0.0",...) is undefined where two or more stacks are in use.
+     * If the socket has not been opened, returns SOCKET_ERROR_NULL_PTR
      *
      * @param[in] address The string representation of the address to bind
      * @param[in] port The local port to bind
      * @return SOCKET_ERROR_NONE on success, or an error code on failure
      */
     virtual socket_error_t bind(const char *address, const uint16_t port);
+
     /**
      * bind(const SocketAddr *, const uint16_t) is the same as bind(const char *, const uint16_t),
-     * except that the address passed in is a SocketAddr.
+     * except that the address passed in is a SocketAddr. If the supplied address is NULL then
+     * SOCKET_ERROR_BAD_ADDRESS is returned.
```

Comment:
Why return BAD_ADDRESS instead of NULL_PTR? Everywhere else in the API, NULL_PTR is returned when a nullptr is passed in.